include/deal.II/matrix_free/operators.h

```diff
@@ -158,7 +158,7 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
-     * Get read access to diagonal of this operator.
+     * Get read access to the inverse diagonal of this operator.
```

Comment:
While looking at this code, I wonder whether this class should actually hold the vector for the diagonal entries. I would prefer if there is simply a function that computes it, say a pure virtual function `compute_matrix_diagonal(VectorType&) const`. I know that this disallows using the `el(i,i)` method. On the other hand, if we plan to introduce a `DiagonalMatrix` class as dicussed in #3223 that class should hold the vector in my opinion. But then we would hold the diagonal twice, which can only lead to confusion. Finally, the inverse diagonal does not logically belong to the operator. (It would need some adjustments to `PreconditionChebyshev` but I have planned that anyway.)